### PR TITLE
Check for and install firmware updates

### PR DESCRIPTION
**Pull request by AlCutter**
This PR enables self-updates on the device.

It's deliberately very aggressive in order to test, we'll dial it down in a bit.

Related changes for CloudBuild in #155

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,11 +45,19 @@ ENTRY_POINT := _rt0_arm_tamago
...
-                  -X 'main.RestDistributorBaseURL=${REST_DISTRIBUTOR_BASE_URL}'"
+                  -X 'main.RestDistributorBaseURL=${REST_DISTRIBUTOR_BASE_URL}' \
+                  -X 'main.updateBinariesURL=${FT_BIN_URL}' \
+                  -X 'main.updateLogURL=${FT_LOG_URL}' \
+                  -X 'main.updateLogOrigin=${FT_LOG_ORIGIN}' \
+                  -X 'main.updateLogVerifier=$(shell cat ${LOG_PUBLIC_KEY})' \
+                  -X 'main.updateAppletVerifier=$(shell cat ${APPLET_PUBLIC_KEY})' \
+                  -X 'main.updateOSVerifier1=$(shell cat ${OS_PUBLIC_KEY1})' \
+                  -X 'main.updateOSVerifier2=$(shell cat ${OS_PUBLIC_KEY2})' \
+				  "
```
Comment (mhutchinson) [inline Makefile:60]: nit: what's that double quote doing all the way out over there?

File: trusted_applet/main.go [hunk 2]
```diff
@@ -230,7 +230,32 @@ func runWithNetworking(ctx context.Context) error {
...
+		return fmt.Errorf("failed to create updater: %v", err)
+	}
+
+	// TODO(al): figure out where & how frequently we should be doing this.
+	// For now, since we're still developing/testing this, we'll be very aggressive
+	// checking for and installing updates.
+	go func() {
+		for {
+			select {
+			case <-time.After(5 * time.Second):
```
Comment (mhutchinson) [inline trusted_applet/main.go:244]: Consider using a ticker here instead. ISTR that time.After causes more garbage but I don't have time to go check my citation right now.

**Review by mhutchinson — COMMENTED** [review]
Comments for now. Will review trusted_applet/update.go tomorrow.

File: Makefile (hunk 1 above)
Comment (AlCutter) [reply Makefile:60]: It was probably on a voyage of self-discovery, pondering the eternal dilemma of tabs vs. spaces...

**Review by mhutchinson — APPROVED** [review]

**Pull request merged**